miner/worker.go

```diff
@@ -284,7 +284,9 @@ func (self *worker) wait() {
 					glog.V(logger.Error).Infoln("Invalid block found during mining")
 					continue
 				}
-				if err := core.ValidateHeader(self.eth.BlockProcessor().Pow, block.Header(), parent.Header(), true, false); err != nil && err != core.BlockFutureErr {
+
+				auxValidator := self.eth.BlockChain().AuxValidator()
```

Comment:
Although we can call the `BlockChain` method as `AuxValidator`, I think the return value should be called `pow` for now here, as header validation accepts that, not some arbitrary "auxiliary validator". We can change it later after it clears up a bit exactly how we want to do this. My 2c.